packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -371,6 +390,16 @@ export function TransferPanel() {
         if (!withdrawalConfirmation) return
       }
 
+      // confirm if the user is certain about the custom destination address, especially if it matches the connected SCW address.
+      // this ensures that user funds do not end up in the destination chain’s address that matches their source-chain wallet address, which they may not control.
+      if (
+        isSmartContractWallet &&
+        isDepositMode &&
+        areSenderAndCustomDestinationAddressesEqual
+      ) {
+        await confirmCustomDestinationAddressForSCWallets()
```

Comment:
Same on L626. Currently if user click on cancel or close, we proceed as if user click on confirm. We should check for the returned value here

Reply:
Updated.